### PR TITLE
w3m: 0.5.3+git20190105 -> 0.5.3+git20220429

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

* Upgrade from 0.5.3+git20190105 to 0.5.3+git20220429
* Add anthonyroussel to maintainers
* Removed `no-x11.patch` in favor of `--without-x` flag
  * See https://github.com/tats/w3m/blob/master/ChangeLog#L723

See changelog: https://github.com/tats/w3m/blob/master/ChangeLog

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by anthonyroussel** [comment]
This PR supersedes my failure with this one: https://github.com/NixOS/nixpkgs/pull/188219

(My apologies again for all these notifications this old PR may have sent, I switched the base branch to `staging` branch without rebasing my work on it :scream:)

Switched to the `staging` base branch because of the mass rebuilds.

**Review by SuperSandro2000 — COMMENTED** [review]
Some other things which should be cleaned up:
- ``? null`` in the inputs is not required when enabling features with booleans.
- ``with lib`` should be removed and functions from lib should be prefixed with ``lib.``.

**Comment by anthonyroussel** [comment]
> Some other things which should be cleaned up:
> 
> * `? null` in the inputs is not required when enabling features with booleans.
> * `with lib` should be removed and functions from lib should be prefixed with `lib.`.

@SuperSandro2000 Hello,

Thanks for the review.

I removed the `? null` from the inputs, the global `with lib` at the top of the file, then prefixed functions using lib with `lib.`.

File: pkgs/applications/networking/browsers/w3m/default.nix [hunk 1]
```diff
@@ -12,8 +12,6 @@ assert graphicsSupport -> imlib2 != null;
 assert x11Support -> graphicsSupport && libX11 != null;
 assert mouseSupport -> gpm-ncurses != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/w3m/default.nix:13]: I forgot to mention: those asserts can now also be deleted

**Review by SuperSandro2000 — APPROVED** [review]
otherwise LGTM

File: pkgs/applications/networking/browsers/w3m/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/applications/networking/browsers/w3m/default.nix]: Removed!

**Pull request merged**